app/pages/admin.js

```diff
@@ -1,25 +1,49 @@
-import Head from "next/head";
-import Image from "next/image";
-import styles from "../styles/Home.module.css";
 import React, { useState, useEffect } from "react";
 import Link from "next/link";
-import { stringify } from "gray-matter";
 
 export default function Admin() {
-  const [links, setLinks] = useState([{id: 0, name: "", link: ""}]);
-  const [amountOfLinks, setAmount] = useState(0);
- 
-  const clickAddLink = () => {
-    setAmount(amountOfLinks + 1); 
-    setLinks(links => [...links, {id: 0, name: "", link: ""}]);
-    }
- 
-    function change (index)
-    {
-      links[index].name = document.getElementById("name");
-    }
-  const [Prev, setPrev] = useState(false);
-  const handleClickPrev = () => setPrev(!Prev);
+  const [link, setLink] = useState({ name: "", link: "" });
+  const [links, setLinks] = useState(new Map());
+
+  const clickAddLink = () => addToMap(links.size + 1, link);
+  const deleteOnMap = (key) => {
+    setLinks((prev) => {
+      const newState = new Map(prev);
+      newState.delete(key);
+      return newState;
+    });
+  };
+  
+ const resetLink = () =>
+ {
+  setLink({ ...link, name: '' })
+  setLink({ ...link, link: '' })
+ }
+
+  const getName = (key) =>
+  { 
+    let nameFromKey = links.get(key)
+    return nameFromKey.name
+  }
+
+  const getLink = (key) =>
+  { 
+    let linkFromKey = links.get(key)
+    return linkFromKey.link
+  }
+  const addToMap = (key, value) => {
+    setLinks((prev) => new Map([...prev, [key, value]]));
+  };
+  const updateMap = (key, value) => {
+    setLinks((prev) => new Map([...prev, [key, value]]));
+  };
+
+  const clear = () => {
+    setLinks((prev) => new Map(prev.clear()));
+  };
+
+  const [prev, setPrev] = useState(false);
+  const handleClickPrev = () => setPrev(!prev);
```

Comment:
Since you're not wrapping this in `useCallback`, use the memory safe version of the state setter. [Reference](https://reactjs.org/docs/hooks-reference.html#functional-updates)
```suggestion
  const handleClickPrev = () => setPrev(p => !p);
```